core/soc/timer.go

```diff
@@ -68,14 +67,32 @@ func (lp *Timer) DemandActive() bool {
 		power *= lp.current / lp.GetMaxCurrent()
 	}
 
+	se := lp.SocEstimator()
+	if se == nil {
+		lp.log.WARN.Printf("target charging: not possible")
+	return false
+	}
+
 	// time
-	remainingDuration := se.RemainingChargeDuration(power, lp.SoC)
+	remainingDuration := se.AssumedChargeDuration(lp.SoC, power)
 	lp.finishAt = time.Now().Add(remainingDuration).Round(time.Minute)
 
+	lp.log.DEBUG.Printf("estimated charge power: %vW", power)
+	lp.log.DEBUG.Printf("estimated charge duration: %v to %v%%", remainingDuration.Round(time.Minute), lp.SoC)
+	lp.log.DEBUG.Printf("projected start: %v", lp.Time.Add(-remainingDuration))
+	if lp.active {
+		lp.log.DEBUG.Printf("projected end (when now starting): %v", lp.finishAt)
+		lp.log.DEBUG.Printf("desired finish time: %v", lp.Time)
+	}
+	lp.log.DEBUG.Printf("starting in: %v", time.Until(lp.Time.Add(-remainingDuration)).Round(time.Second))
```

Comment:
ist "starting in" nicht doppelt zu "projected start"? Die 2x if sehen doof aus.

Reply:
ja, kann ich aufräumen. Ich hätte in einem ersten Schritt eher mehr Debugausgaben drinnen gelassen um bei Fehler auch was sagen zu können und das dann später aufgeräumt. Der Grund warum beide drinnen waren war weil ich befüchtet habe dass es ein Problem mit Zeitzonen geben kann und daher hätte ich mir den Startpunkt und die Zeitdauer angezeigt. Ich räume das noch etwas auf.